### PR TITLE
DD Store Cleanup

### PR DESCRIPTION
**Pull request by ianna**
* If the Store holds raw pointers, make sure they are cleaned up
* Address the comments from @kpedro88 on the previous PRs: https://github.com/cms-sw/cmssw/pull/24313, https://github.com/cms-sw/cmssw/pull/24263, https://github.com/cms-sw/cmssw/pull/24326

### CODE REVIEW

**Comment by ianna** [comment]
please test

**Comment by kpedro88** [comment]
@ianna not all comments are addressed here yet:
https://github.com/cms-sw/cmssw/pull/24263#discussion_r211392285
https://github.com/cms-sw/cmssw/pull/24263#discussion_r211398116
https://github.com/cms-sw/cmssw/pull/24263#discussion_r211398188

**Comment by kpedro88** [comment]
and also:
https://github.com/cms-sw/cmssw/pull/24313#discussion_r210707857

**Comment by ianna** [comment]
hold

File: DetectorDescription/Core/plugins/DDLinear.cc [hunk 1]
```diff
@@ -104,7 +105,7 @@ DDLinear::execute( DDCompactView& cpv )
     LogDebug( "DDAlgorithm" ) << "DDLinear: Creating a new "
 			      << "rotation: IdentityRotation for " << ddname;
 	
-    rotation = DDrot( "IdentityRotation", new DDRotationMatrix());
+    rotation = DDrot( "IdentityRotation", std::make_unique< DDRotationMatrix >());
```
Comment (ianna) [inline DetectorDescription/Core/plugins/DDLinear.cc]: @kpedro88 - FYI

File: DetectorDescription/Core/src/DDCompactView.cc [hunk 2]
```diff
@@ -112,28 +109,29 @@ void DDCompactView::swap( DDCompactView& repToSwap ) {
 }
 
 DDCompactView::DDCompactView()
-  : rep_(new DDCompactViewImpl),
+  : rep_( new DDCompactViewImpl ),
```
Comment (ianna) [inline DetectorDescription/Core/src/DDCompactView.cc]: @kpedro88 - I'm not sure I prefer the verbose version:
```
: rep_( std::make_unique<DDCompactViewImpl>() ),
```
IMHO, using ```new``` is perfectly fine here
@Dr15Jones - please, comment

File: DetectorDescription/OfflineDBLoader/src/DDCoreToDDXMLOutput.cc [hunk 3]
```diff
@@ -388,7 +388,7 @@ DDCoreToDDXMLOutput::rotation( const DDRotation& rotation, std::ostream& xos,
       {
          rotName = rotn;
          std::cout << "about to try to make a new DDRotation... should fail!" << std::endl;
-         DDRotation rot( DDName(rotn), const_cast<DDRotationMatrix*>(rotation.rotation()));
+         DDRotation rot( DDName(rotn), std::make_unique<DDRotationMatrix>( rotation.rotation()));
```
Comment (ianna) [inline DetectorDescription/OfflineDBLoader/src/DDCoreToDDXMLOutput.cc]: @kpedro88 - FYI

**Comment by ianna** [comment]
@kpedro88 - FYI, @Dr15Jones's https://github.com/cms-sw/cmssw/pull/24313#discussion_r210707857 has been taken in consideration for future work on DDName. One step at a time :-)

**Comment by ianna** [comment]
please test

File: DetectorDescription/Core/src/DDCompactView.cc (hunk 2 above)
Comment (kpedro88) [reply DetectorDescription/Core/src/DDCompactView.cc]: It makes it clearer that smart pointers are being used. A bare `new` looks like a potential memory leak, even though it isn't in this case.

Comment (Dr15Jones) [reply DetectorDescription/Core/src/DDCompactView.cc]: I also prefer the use of `std::make_unique` for the same reasons as Kevin.

**Comment by mrodozov** [comment]
please test

File: DetectorDescription/Core/src/DDCompactView.cc (hunk 2 above)
Comment (ianna) [reply DetectorDescription/Core/src/DDCompactView.cc]: Fixed in https://github.com/cms-sw/cmssw/pull/24378/commits/66ed9285c07525a08e43a0ee2c3a640d72ea95ac

**Comment by ianna** [comment]
unhold

**Comment by ianna** [comment]
+1

**Comment by andrius-k** [comment]
+1

**Comment by kpedro88** [comment]
+upgrade

**Comment by civanch** [comment]
+1

**Comment by ianna** [comment]
@alja - please, review

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
+1

@kpedro88 - FYI, it's just a rebase to remove unneeded signatures

**Comment by kpedro88** [comment]
+1

**Pull request merged**